docs/DafnyRef/Attributes.md

```diff
@@ -344,136 +302,206 @@ recursion was explicitly requested.
-       attribute--it does not try to verify that the function really is an
-       identity function.
-
-  ---- On variables ----------------------------------------------------------
-
-     {:existential true}
-       Marks a global Boolean variable as existentially quantified. If
-       used in combination with option /contractInfer Boogie will check
-       whether there exists a Boolean assignment to the existentials
-       that makes all verification conditions valid.  Without option
-       /contractInfer the attribute is ignored.
-
-  ---- On assert statements --------------------------------------------------
-
-     {:subsumption n}
-       Overrides the /subsumption command-line setting for this assertion.
-
-     {:split_here}
-       Verifies code leading to this point and code leading from this point
-       to the next split_here as separate pieces.  May help with timeouts.
-       May also occasionally double-report errors.
-
-  ---- The end ---------------------------------------------------------------
 
+The quantifier is translated into the following Boogie:
+
+```
+(forall q$heap#8: Heap, u#5: int ::
+    {:heapQuantifier}
+    $IsGoodHeap(q$heap#8) && ($Heap == q$heap#8 || $HeapSucc($Heap, q$heap#8))
+       ==> $Unbox(Apply1(TInt, TInt, f#0, q$heap#8, $Box(u#5))): int == u#5 + r#0);
 ```
 
-However a scan of Boogie's sources shows it checks for the
+What this is saying is that the quantified expression, `f(u) == u + r`,
+which may depend on the heap, is also valid for any good heap that is either the
+same as the current heap, or that is derived from it by heap update operations.
+
+### 22.5.2. `{:induction}` {#sec-induction-quantifier}
+See [`{:induction}`](#sec-induction).
+
+### 22.5.3. `{:layerQuantifier}`
+When Dafny is translating a quantified expression, if it has
+a `{:layerQuantifier}` attribute an additional quantifier
+variable is added to the quantifier bound variables.
+This variable as the predefined _LayerType_.
+A `{:layerQuantifier}` attribute may be placed on a quantifier expression.
+Translation of Dafny into Boogie defines a _LayerType_ which has defined zero and
+successor constructors.
+
+The Dafny source has the comment that "if a function is recursive,
+then make the reveal lemma quantifier a layerQuantifier."
+And in that case it adds the attribute to the quantifier.
+
+There is no explicit user of the `{:layerQuantifier}` attribute
+in the Dafny tests. So I believe this attribute is only used
+internally by Dafny and not externally.
+
+TODO: Need more complete explanation of this attribute.
+Dafny issue [35](https://github.com/Microsoft/dafny/issues/35) tracks
+further effort for this attribute.
+
+### 22.5.4. `{:trigger}` {#sec-trigger}
+Trigger attributes are used on quantifiers and comprehensions.
+They are used for the verifier to know models of expressions that enable to instantiate the quantifier,
+so that it is not instantiated with anything.
```

Comment:
This sentence is a little unclear.

Reply:
I replaced this sentence with a detailed example.